game/src/select_cards.rs

```diff
@@ -49,13 +52,343 @@ fn get_crib_cards(hand: &[Card], held_cards: &[Card]) -> Vec<Card> {
+    //  a 2, then it would be common for that person to also have a 3.  they played the 2 hoping that a card of value 10 would be played and then they
+    //  could play a 3 - 15 for 2 points!.  This algorithm will always play a 2 to pair the 2 if it is possible.  Another strategy would be to play a 4
+    //  or an Ace (*never* a 3 - because you don't want the count to be 5 because of how many cards are available with value=10).  in this case, the opponent
+    //  would play the 3 and get a run of 3 for 3 points.  But then you could build on the run and get 4 points (assuming you had the right cards)
+    //
+    //  one of the things I want to explore is defeating the "get a 15" strategy above. the other common strategy (which we use below) is to start with a card
+    //  that you have a pair...the hope is the apponent pairs the card, so you can get 3 of a kind (e.g. play a 4, opponent plays a 4, you play your second 4)
+    //  would it be better to not pair the card and give up the two points?  over the course of the game, stopping 3 of a kind might be worth it.
+    //
+    //  this is an area ripe for innovation!
+    //
+
+    //
+    //  Note: I need the card to play for the max score - i tried to do this in a variety of ways using map/fold and I could get the max score, but the card
+    //        that generated the max score was lost in the map.  i gave up and did it via iteration.
+
+    let mut max: i32 = -1;
+    let mut card_to_play = Card::new(Rank::Unknown, Suit::Unknown);
+    let mut playable_cards: Vec<Card> = Vec::new();
+    for potential_card in cards_left.as_slice() {
+        if let Ok(s) = score_counting_cards_played(counted_cards.as_slice(), *potential_card) {
+            if max < s.total_score as i32 {
+                // innovation idea:  if you can play 2 cards and get the same score, which one should you play?
+                max = s.total_score as i32;
+                card_to_play = *potential_card;
+            }
+
+            playable_cards.push(*potential_card);
+        }
+    }
+
+    if playable_cards.is_empty() {
+        // this means we have no valid cards to play
+        return Err("Go".into());
+    }
+    //
+    //  get the most points - innovate here and pick the *best* one, not necessarily the one with most points!
+    if max > 0 {
+        return Ok(card_to_play);
+    }
+    //
+    //  we can play, but we can't get points.
+
+    //
+    //  find all combinations of 2 cards in the cards I have left...
+    let two_card_combi = all_combinations_of_size(playable_cards, 2, 2);
+    //
+    //  we only want to interate through the 2 card combinations once, so
+    //  we set a strategic weight to pick between the various scenarios and
+    //  set the card_to_play when we find one
+    //
+    let mut strategic_weight = 0;
+    let mut card_to_play: Card = Card::new(Rank::Unknown, Suit::Unknown);
+
+    for mut cards in two_card_combi {
+        if cards.len() != 2 {
```

Comment:
This is a weird panic. At some point, we should trust the code above, that we own, to do what it says it does.